lib/utils.js

```diff
@@ -565,12 +565,18 @@ function uniq(a) {
   if (!a) {
     return uniqArray;
   }
+  a = a.map(element => {
+    if (typeof element === 'object')
+      return String(element);
```

Comment:
Won't this return `[object Object]`?

Reply:
If i remove String(element) i get bsonType: ['_bsontype', 'id']. In this way i get a simple id String.
I'll put the tests as soon as possible, meantime I have implemented a simple test on this repo: https://github.com/BlueRoboto/loopback-sandbox.